subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -699,6 +797,45 @@ static u16_t atou16(u8_t *buf, u16_t buflen, u16_t *len)
 	return val;
 }
 
+static int atof32(const char *input, float32_value_t *out)
+{
+	char *pos, *end, buf[24];
+	long int val;
+	s32_t base = 1000000;
+
+	if (!input || !out) {
+		return -EINVAL;
+	}
+
+	strncpy(buf, input, sizeof(buf) - 1);
+	buf[sizeof(buf) - 1] = '\0';
+
+	pos = strchr(buf, '.');
+	if (pos) {
+		*pos = '\0';
+	}
+
+	val = strtol(buf, &end, 10);
+	if (*end || val > INT_MAX || val < INT_MIN) {
```

Comment:
LONG_MIN and INT_MIN are usually the same on 32-bit (at least that seems to be the case in the minimal libc); same goes for the MAX values.  So it's better to check for errno here:

```c
errno = 0;
val = strtol(buf, &end, 10);
if (errno || *end) {
    return -EINVAL;
}
```

(Zeroing out `errno` before calling `strtol()` is necessary since it does not reset if no error was detected.)